pandas/tests/test_resample.py

```diff
@@ -2564,6 +2564,22 @@ def test_upsample_daily_business_daily(self):
         expected = ts.asfreq('H', how='s').reindex(exp_rng)
         assert_series_equal(result, expected)
 
+
+    # GH12351
+    def test_resample_business_hourly(self):
+        rng = pd.date_range(start='2017-05-18 00:00:00',
+                            end='2017-05-19 23:00:00',
+                            freq='H')
+        expected_rng = pd.date_range(start='2017-05-18 00:00:00',
+                                    end='2017-05-19 23:00:00',
+                                    freq='BH')
+        ts = Series(1, index=rng)
+        result = ts.asfreq('BH')
+        expected_ts = Series(1, index=expected_rng)
+
+        assert_series_equal(result, expected_ts)
+
```

Comment:
Could you add a test against a `CustomBusinessHour` (I've never used one, but can help figure it out if you need).